server/service/src/rnr_form/finalise.rs

```diff
@@ -90,3 +102,89 @@ impl From<RepositoryError> for FinaliseRnRFormError {
         FinaliseRnRFormError::DatabaseError(error)
     }
 }
+
+fn create_internal_order(
+    ctx: &ServiceContext,
+    rnr_form: RnRFormRow,
+) -> Result<(), FinaliseRnRFormError> {
+    // Internal Orders are known as requisitions in the code base
+    let requisition_row = RequisitionRow {
+        id: uuid(),
+        user_id: Some(ctx.user_id.clone()),
+        requisition_number: next_number(
+            &ctx.connection,
+            &NumberRowType::RequestRequisition,
+            &ctx.store_id,
+        )?,
+        name_link_id: rnr_form.name_link_id,
+        store_id: rnr_form.store_id,
+        r#type: RequisitionType::Request,
+        status: RequisitionStatus::Sent,
+        created_datetime: Utc::now().naive_utc(),
+        sent_datetime: None,
+        finalised_datetime: None,
+        expected_delivery_date: None,
+        colour: None,
+        comment: Some("Automatically created from R&R Form".to_string()),
+        their_reference: None,
+        max_months_of_stock: 0.0,
+        min_months_of_stock: 0.0,
+        approval_status: None,
+        linked_requisition_id: None,
+        program_id: Some(rnr_form.program_id),
+        period_id: Some(rnr_form.period_id),
+        order_type: None, // Should we capture this in the RnR form?
+    };
+
+    let requisition_repo = RequisitionRowRepository::new(&ctx.connection);
+    requisition_repo.upsert_one(&requisition_row)?;
+
+    // Get R&R Form lines and create requisition lines
+
+    let rnr_form_line_repo = RnRFormLineRowRepository::new(&ctx.connection);
+    let requisition_line_repo = RequisitionLineRowRepository::new(&ctx.connection);
+    let item_repo = ItemRowRepository::new(&ctx.connection);
+
+    let rnr_form_lines = rnr_form_line_repo.find_many_by_rnr_form_id(&rnr_form.id)?;
+
+    // Find all the item names
+    let item_ids = rnr_form_lines
+        .iter()
+        .map(|rnr_form_line| rnr_form_line.item_id.clone())
+        .collect();
+
+    let items = item_repo.find_many_by_id(&item_ids)?;
+
+    let item_map = items
+        .into_iter()
+        .map(|item| (item.id.clone(), item))
+        .collect::<std::collections::HashMap<String, _>>();
+
+    // Loop through the rnr lines and create requisition lines
+
+    for rnr_form_line in rnr_form_lines {
+        let requisition_line = RequisitionLineRow {
+            id: uuid(),
+            requisition_id: requisition_row.id.clone(),
+            item_link_id: rnr_form_line.item_id.clone(),
+            item_name: item_map
+                .get(&rnr_form_line.item_id)
+                .map(|item| item.name.clone())
+                .unwrap_or_default(),
+            requested_quantity: rnr_form_line.requested_quantity,
+            suggested_quantity: rnr_form_line.requested_quantity, // TODO This is user enterable, should we also re-calc this?
```

Comment:
Not quite sure what the suggested_quantity should be, I think it probably should be the calculated value (but user can over write it) Do I re-calc here?